src/docker.rs

```diff
@@ -179,3 +194,232 @@ pub fn image(toml: Option<&Toml>, target: &Target) -> Result<String> {
 
     Ok(image)
 }
+
+fn docker_read_mount_paths() -> Result<Vec<MountDetail>> {
+    let hostname = if let Ok(v) = env::var("HOSTNAME") {
+        Ok(v)
+    } else {
+        Err("HOSTNAME environment variable not found")
+    }?;
+
+    let docker_path = which::which(DOCKER)?;
+    let mut docker: Command = {
+        let mut command = Command::new(docker_path);
+        command.arg("inspect");
+        command.arg(hostname);
+        command
+    };
+
+    let output = docker.run_and_get_stdout(false)?;
+    let info = if let Ok(val) = serde_json::from_str(&output) {
+        Ok(val)
+    } else {
+        Err("failed to parse docker inspect output")
+    }?;
+
+    dockerinfo_parse_mounts(&info)
+}
+
+fn dockerinfo_parse_mounts(info: &serde_json::Value) -> Result<Vec<MountDetail>> {
+    let mut mounts = dockerinfo_parse_user_mounts(info);
+    let root_info = dockerinfo_parse_root_mount_path(info)?;
+    mounts.push(root_info);
+    Ok(mounts)
+}
+
+fn dockerinfo_parse_root_mount_path(info: &serde_json::Value) -> Result<MountDetail> {
+    let driver_name = info
+        .pointer("/0/GraphDriver/Name")
+        .and_then(|v| v.as_str())
+        .ok_or("No driver name found")?;
+
+    if driver_name == "overlay2" {
+        let path = info
+            .pointer("/0/GraphDriver/Data/MergedDir")
+            .and_then(|v| v.as_str())
+            .ok_or("No merge directory found")?;
+
+        Ok(MountDetail {
+            source: PathBuf::from(&path),
+            destination: PathBuf::from("/"),
+        })
+    } else {
+        Err(format!("want driver overlay2, got {}", driver_name).into())
+    }
+}
+
+fn dockerinfo_parse_user_mounts(info: &serde_json::Value) -> Vec<MountDetail> {
+    info.pointer("/0/Mounts")
+        .and_then(|v| v.as_array())
+        .map(|v| {
+            let make_path = |v: &serde_json::Value| PathBuf::from(&v.as_str().unwrap());
+            let mut mounts = vec![];
+            for details in v {
+                let source = make_path(&details["Source"]);
+                let destination = make_path(&details["Destination"]);
+                if source != destination {
```

Comment:
@urso I am not sure to understand why this check is needed?
Actually, I have tried to run latest cross from github with docker-in-docker and if I mount my rust project as `-v $PWD:$PWD -w $PWD`, `source == destination` so it is not mounted in the second stage docker and `/project` is empty.
I might open an issue to discuss it further if you want.
Thank you.

Reply:
This is just a 'translation' table replacing the prefix that matches 'destination' with the prefix that matches 'source'. If `source == destination`, then this will be a noop.

If you open a shell in your development/CI container you can check the environment via `$ echo $CROSS_DOCKER_IN_DOCKER; docker inspect $HOSTNAME`. Running `cross <command> -v` will also print the docker commands as they will be executed. Very helpful for debugging.

Reply:
Thanks, I did some debugging yesterday and by removing this check [in my fork](https://github.com/romainreignier/cross/commit/52757396d812291b18cfece817fb8895da3dc934) allowed me to use cross in a docker ran by Jenkins.

The issue here is that Jenkins mount the current workspace with the same path in Docker with
```
-v /path/to/my/workspace:/path/to/my/workspace
```

So `source == destination` but it is not a noop, it allows to mount files from host to the second docker stage.

I have managed to reproduce with:
```bash
# On host, create a file in current directory
touch testfile
# start a first docker stage
docker run -it --rm -v /var/run/docker.sock:/var/run/docker.sock -v $PWD:$PWD -w $PWD debian:buster bash
# in the first docker stage, install docker
apt update && apt install -y --no-install-recommends docker.io
# check that the test file is there
ls
# it is here, OK
# get the merged path
merge_path=$(docker inspect `hostname` --format '{{.GraphDriver.Data.MergedDir}}')
# start the second docker stage
docker run -it --rm -v $merge_path/$PWD:/project debian:buster bash
# in the second docker stage, check that the test file is here
ls /project
# there is no file, NOK
# exit the second docker
exit
# new start with classic mount
docker run -it --rm -v $PWD:/project debian:buster bash
ls /project
# the testfile is here, OK
```

Reply:
Oh, I see. If `source == destination` for mounted dirs, then the replacement will swap out the root path `/` with the containers merge path. Thanks to default mounting rules the `/project` will be empty (mounts are not passed through by default). This is why we need the path translation rules in the first place.

You are right, we should remove the if condition and always add the mount infos. I'm no maintainer, nor regular contributor to `cross`, but would be nice if you can create a PR removing this check.

Reply:
Ok, thanks for having a look to it. I will create a PR.